File: user-service/src/main/java/com/example/user/dto/KakaoUserInfoResponse.java
package com.example.user.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class KakaoUserInfoResponse {

    private Long id;

    @JsonProperty("kakao_account")
    private KakaoAccount kakaoAccount;

    @Getter
    @NoArgsConstructor
    public static class KakaoAccount {
        private String email;
        private Profile profile;

        @Getter
        @NoArgsConstructor
        public static class Profile {
            private String nickname;

            @JsonProperty("profile_image_url")
            private String profileImageUrl;
        }
    }

    public String getNickname() {
        if (kakaoAccount != null && kakaoAccount.getProfile() != null) {
            return kakaoAccount.getProfile().getNickname();
        }
        return null;
    }

    public String getEmail() {
        if (kakaoAccount != null) {
            return kakaoAccount.getEmail();
        }
        return null;
    }

    public String getProfileImageUrl() {
        if (kakaoAccount != null && kakaoAccount.getProfile() != null) {
            return kakaoAccount.getProfile().getProfileImageUrl();
        }
        return null;
    }
}

File: apps/user-service/src/auth/kakao/kakao-user-info-response.ts
export interface KakaoUserInfoResponse {
  id: number;
  kakao_account?: {
    email?: string;
    profile?: {
      nickname?: string;
      profile_image_url?: string;
    };
  };
}

export function getNickname(info: KakaoUserInfoResponse): string | null {
  return info.kakao_account?.profile?.nickname ?? null;
}

export function getEmail(info: KakaoUserInfoResponse): string | null {
  return info.kakao_account?.email ?? null;
}

export function getProfileImageUrl(info: KakaoUserInfoResponse): string | null {
  return info.kakao_account?.profile?.profile_image_url ?? null;
}
